doc/nrf/releases_and_maturity/migration/migration_guide_2.9.rst

```diff
@@ -61,4 +61,156 @@ Libraries
+            * The :kconfig:option:`CONFIG_DOWNLOAD_CLIENT_COAP_BLOCK_SIZE` Kconfig option with the :kconfig:option:`CONFIG_DOWNLOADER_COAP_BLOCK_SIZE_512` Kconfig option.
+
+         * Remove:
+
+            * The :kconfig:option:`CONFIG_DOWNLOAD_CLIENT_BUF_SIZE` Kconfig option.
+            * The :kconfig:option:`CONFIG_DOWNLOAD_CLIENT_HTTP_FRAG_SIZE` Kconfig option.
+            * The :kconfig:option:`CONFIG_DOWNLOAD_CLIENT_RANGE_REQUESTS` Kconfig option.
+            * The :kconfig:option:`CONFIG_DOWNLOAD_CLIENT_CID` Kconfig option.
+
+         * Add:
+
+            * The :kconfig:option:`CONFIG_DOWNLOADER_TRANSPORT_COAP` Kconfig option to enable CoAP support.
+            * The :kconfig:option:`CONFIG_NET_IPV4` Kconfig option to enable IPv4 support.
+            * The :kconfig:option:`CONFIG_NET_IPV6` Kconfig option to enable IPv6 support.
+
+     #. Replace header files:
+
+        * Remove:
+
+          .. code-block:: C
+
+           #include <net/download_client.h>
+
+        * Add:
+
+          .. code-block:: C
+
+           #include <net/downloader.h>
+
+     #. Replace download client initialization:
+
+         * Remove:
+
+          .. code-block:: C
+
+            static struct download_client dlc;
+            static int callback(const struct download_client_evt *event);
+
+            download_client_init(&dlc, callback)
+
+         * Add:
+
+         .. code-block:: C
+
+            static struct downloader dl;
+            static int callback(const struct downloader_evt *event);
+            static char dl_buf[2048]; /* Use buffer size set by CONFIG_DOWNLOAD_CLIENT_BUF_SIZE previously */
+            static struct downloader_cfg dl_config = {
+               .callback = callback,
+               .buf = dl_buf,
+               .buf_size = sizeof(dl_buf),
+            };
+
+            downloader_init(&dl, &dl_config);
+
+     #. Update download client callback:
+
+         * Replace:
+
+            * :c:enum:`DOWNLOAD_CLIENT_EVT_FRAGMENT` event with :c:enum:`DOWNLOADER_EVT_FRAGMENT`.
+            * :c:enum:`DOWNLOAD_CLIENT_EVT_ERROR` event with :c:enum:`DOWNLOADER_EVT_ERROR`.
+            * :c:enum:`DOWNLOAD_CLIENT_EVT_DONE` event with :c:enum:`DOWNLOADER_EVT_DONE`.
+            * :c:enum:`DOWNLOAD_CLIENT_EVT_DONE` event with :c:enum:`DOWNLOADER_EVT_DONE`.
+
+         * Remove:
+
+            * :c:enum:`DOWNLOAD_CLIENT_EVT_CLOSED` event.
+
+         * Add:
+
+            * :c:enum:`DOWNLOADER_EVT_STOPPED` event.
+            * :c:enum:`DOWNLOADER_EVT_DEINITIALIZED` event.
+
+     #. Server connect and disconnect:
+
+         * The :c:func:`download_client_disconnect` function is not ported to the new downloader.
+           The client is expected to connect when the download begins, and have the ``keep_connection`` flag set in the host configuration if the connection should persist after the download completes or is aborted by calling the :c:func:`downloader_stop` function.
```

Comment:
Is :c:func:`downloader_stop` the correct one? Couldn't find this in the header file.

Reply:
Renamed to `downloader_cancel`.